python/lsst/pipe/base/pipelineBuilder.py

```diff
@@ -55,123 +50,83 @@
-
-        else:
-            # start with empty one
-            pipeline = Pipeline()
-
-        # loop over all pipeline actions and apply them in order
-        for action in args.pipeline_actions:
-
-            if action.action == "new_task":
-
-                self._newTask(pipeline, action.value, action.label, obsPkg, instrument)
-
-            elif action.action == "delete_task":
-
-                self._dropTask(pipeline, action.label)
-
-            elif action.action == "move_task":
-
-                self._moveTask(pipeline, action.label, int(action.value))
-
-            elif action.action == "relabel":
-
-                self._labelTask(pipeline, action.label, action.value)
-
-            elif action.action == "config":
-
-                self._configOverride(pipeline, action.label, action.value)
-
-            elif action.action == "configfile":
-
-                self._configOverrideFile(pipeline, action.label, action.value)
-
         # conditionally re-order pipeline if requested, but unconditionally
         # check for possible errors
-        orderedPipeline = pipeTools.orderPipeline(pipeline, self.taskFactory)
-        if args.order_pipeline:
-            pipeline = orderedPipeline
-
-        return pipeline
+        orderedPipeline = pipeTools.orderPipeline(self._pipeline, self._taskFactory)
+        if order_pipeline:
+            return orderedPipeline
+        else:
+            return self._pipeline
 
-    def _newTask(self, pipeline, taskName, label=None, obsPkg=None, instrument=None):
+    def addTask(self, taskName, label=None, instrument_overrides=False):
         """Append new task to a pipeline.
 
         Parameters
         ----------
-        pipeline : py:class:`Pipeline`
-            Pipeline instance.
         taskName : `str`
             Name of the new task, can be either full class name including
             package and module, or just a class name to be searched in
             known packages and modules.
         label : `str`, optional
             Label for new task, if `None` the n task class name is used as
             label.
-        obsPkg : `str`, optional
-            Name of the package to look for task overrides, if None then
-            overrides are not used.
-        instrument : `str`, optional
-            Instrument name, used for instrument-specific overrides an only if
-            `obsPkg` is not `None`.
+        instrument_overrides : `bool`, optional
+            If `True` then apply instrument/package overrides to task
+            configuration.
         """
+        # need instrument/package name to find overrides
+        obsPkg = None
+        instrument = None
+        if instrument_overrides:
+            # TODO: will need something different for gen3
+            # mapperClass = dafPersist.Butler.getMapperClass(args.input)
+            # instrument = mapperClass.getCameraName()
+            # obsPkg = mapperClass.getPackageName()
+            pass
+
```

Comment:
I would recommend just removing all of the placeholder logic for applying obs-package overrides.  It doesn't do anything yet, and I'm not at all confident that we'll want to do it at this stage anyway.